src/lib/serviceRoutines/statisticsTreat.cpp

```diff
@@ -509,23 +517,37 @@ std::string statisticsTreat
     out += TAG_ADD_STRING("subCache",          listBuffer);
   }
 
+  std::string timingStatString      = timingStatistics(indent2, ciP->outFormat, ciP->apiVersion);
+  bool        timingStat            = (timingStatString != "");
+  bool        threadpool            = strcmp(notificationMode, "threadpool") == 0;
+  bool        commaAfterThreadpool  = timingStat;
+  bool        commaAfterSubCache    = timingStat || threadpool;
+
   out += TAG_ADD_INTEGER("subCacheRefreshs", mscRefreshs, true);
   out += TAG_ADD_INTEGER("subCacheInserts",  mscInserts,  true);
   out += TAG_ADD_INTEGER("subCacheRemoves",  mscRemoves,  true);
   out += TAG_ADD_INTEGER("subCacheUpdates",  mscUpdates,  true);
-  out += TAG_ADD_INTEGER("subCacheItems",    cacheItems,  false);
+  out += TAG_ADD_INTEGER("subCacheItems",    cacheItems,  commaAfterSubCache);
 
   if (strcmp(notificationMode, "threadpool") == 0)
   {
-    out += TAG_ADD_INTEGER("noOfNotificationsQueueIn", QueueStatistics::getIn(), true);
-    out += TAG_ADD_INTEGER("noOfNotificationsQueueOut", QueueStatistics::getOut(), true);
-    out += TAG_ADD_INTEGER("noOfNotificationsQueueReject", QueueStatistics::getReject(), true);
-    out += TAG_ADD_INTEGER("noOfNotificationsQueueSentOK", QueueStatistics::getSentOK(), true);
-    out += TAG_ADD_INTEGER("noOfNotificationsQueueSentError", QueueStatistics::getSentError(), false);
+    out += TAG_ADD_INTEGER("noOfNotificationsQueueIn",        QueueStatistics::getIn(),        true);
+    out += TAG_ADD_INTEGER("noOfNotificationsQueueOut",       QueueStatistics::getOut(),       true);
+    out += TAG_ADD_INTEGER("noOfNotificationsQueueReject",    QueueStatistics::getReject(),    true);
+    out += TAG_ADD_INTEGER("noOfNotificationsQueueSentOK",    QueueStatistics::getSentOK(),    true);
+    out += TAG_ADD_INTEGER("noOfNotificationsQueueSentError", QueueStatistics::getSentError(), commaAfterThreadpool);
+  }
+
+
+  //
+  // Timing statistics ONLY if JSON
+  //
+  if (ciP->outFormat == JSON)
```

Comment:
Does this means that if statistics are rendered in XML timing statistics are not included?

Just to be sure for documentation purpose... In fact, probably we could adopt a more radical approach and enable statistics in JSON format only (let's take into account that XML is officially deprecated as request/response encoding since 0.23.0: https://github.com/telefonicaid/fiware-orion/blob/develop/doc/manuals/deprecated.md).